libs/traits/src/rewards.rs

```diff
@@ -238,11 +238,15 @@ pub mod mock {
 				amount: <Self as AccountRewards<AccountId>>::Balance,
 			) -> DispatchResult;
 
+			// Return type is too messy for clippy, but not messy enough to really clean up
+			#[allow(clippy::type_complexity)]
```

Comment:
I believe we can set the threshold for clippy type  complexity, if we so desire. I feel like these `Result <Self as Trait>::Type, Error>` patterns crop up quite a bit in our codebase, and I don't think its worth making a `TypeOf<Trait>` alias for every single time it comes up.

Reply:
To add context here, the `mockall::mock!` macro forces you to write the extended trait path version (not sure why) instead of `Result<Self::Type, Error>`